iphone/Maps/UI/BottomMenu/Menu/Cells/BottomMenuItemCell.swift

```diff
@@ -33,28 +32,23 @@ class BottomMenuItemCell: UITableViewCell {
     icon.setStyleAndApply(isEnabled ? "MWMBlack" : "MWMGray")
     label.setStyleAndApply(isEnabled ? "blackPrimaryText" : "blackHintText")
   }
-  
-  func configure(imageName: String, title: String) {
-    icon.image = UIImage.init(named: imageName)
-    label.text = title
-    icon.setStyleAndApply("MWMBlue")
-    label.setStyleAndApply("linkBlueText")
-    badgeBackground.isHidden = true
-    badgeSpacingConstraint.constant = 0
-    badgeBackgroundWidthConstraint.constant = 0
-    isEnabled = true
-    isPromo = true
-  }
-  
+
   override func setHighlighted(_ highlighted: Bool, animated: Bool) {
     guard isEnabled else {
       return
     }
     super.setHighlighted(highlighted, animated: animated)
-    if isPromo {
-      label.setStyleAndApply(highlighted ? "linkBlueHighlightedText" : "linkBlueText")
-    } else {
-      label.setStyleAndApply(highlighted ? "blackHintText" : "blackPrimaryText")
+    label.setStyleAndApply(highlighted ? "blackHintText" : "blackPrimaryText")
+  }
+
+  func setState(_ state: TrackRecordingState) {
+    switch state {
+    case .inactive, .error:
```

Comment:
What causes the error state? Should the text be different for it?

Reply:
Error state here is when Location is prohibited (for now it is the only track recording error)